ArjunaJTA/jdbc/tests/classes/com/hp/mwtests/ts/jdbc/basic/ConnectionImpleTest.java

```diff
@@ -125,13 +124,14 @@ public void beforeCompletion() { }
             public void afterCompletion(int status) {
                 try {
                     connectionToTest.isClosed();
+                    connectionToTest.close();
```

Comment:
But connection wasn't really opened. xaDataSource is a mock, so instead of creating a connection it returns a mock connection.
Also message in the catch block is a bit misleading since you've removed close statement.